packages/context/src/context.ts

```diff
@@ -637,6 +686,32 @@ export class Context extends EventEmitter {
     }
   }
 
+  /**
+   * Remove tag index for the given binding
+   * @param binding - Binding object
+   */
+  private removeTagIndexForBinding(binding: Readonly<Binding<unknown>>) {
+    for (const [, bindings] of this.bindingsIndexedByTag) {
+      bindings.delete(binding);
+    }
+  }
+
+  /**
+   * Update tag index for the given binding
+   * @param binding - Binding object
+   */
+  private updateTagIndexForBinding(binding: Readonly<Binding<unknown>>) {
+    this.removeTagIndexForBinding(binding);
+    for (const tag of binding.tagNames) {
+      let bindings = this.bindingsIndexedByTag.get(tag);
+      if (bindings == null) {
+        bindings = new Set();
+        this.bindingsIndexedByTag.set(tag, bindings);
+      }
+      bindings.add(binding);
+    }
+  }
```

Comment:
I am concerned about the ever-growing size of `packages/context/src/context.ts`. Is it possible to move these new functions to a standalone class, e.g. `ContextTagIndexer` and then use methods of that class for event listeners and when looking up bindings by tags?

Reply:
Extracted into `ContextTagIndexer`.